java/ql/lib/semmle/code/java/dispatch/ObjFlow.qll

```diff
@@ -80,6 +80,20 @@ private predicate step(Node n1, Node n2) {
       for.getVariable() = def.(BaseSsaUpdate).getDefiningExpr() and
       for.getExpr() = n1.asExpr()
     )
+    or
+    exists(PatternCase pc |
```

Comment:
These couple of added cases should be superfluous once they are included in `VariableAssign.getSource()`.